### PR TITLE
GoogleDrive API

### PR DESCRIPTION
**Pull request by VlVl**
googleDrive tools and error classes moved from plyo.db
Google credentials removed from `process.env`. It is needed to call `setup` function and to pass config object before any other functions call

### CODE REVIEW

**Comment by VlVl** [comment]
[Implement web-to-pdf service Proof-of-concept](https://trello.com/c/ruRCW01l/1081-implement-web-to-pdf-service-proof-of-concept)

File: lib/googleDrive/googleAuth.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import GoogleAuth from 'google-auth-library';
+
+export default function (googleApi) {
```
Comment (jifeon) [inline lib/googleDrive/googleAuth.js]: avoid anonymous functions, give it a name. `default export` should have the same name as a file name

File: lib/googleDrive/index.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+/**
+ * @param {http.Request} req
+ * @param {string} fieldName - where to look for file
+ * @param {object} options
+ * @param {string} options.folder - directory to upload to
+ * @returns {Promise.<GoogleDriveFileResource, Error>}
+ */
+export async function upload(req, fieldName, options) {
+  const parentDirectoryId = await getFolderId(options.folder);
```
Comment (jifeon) [inline lib/googleDrive/index.js]: would be nice to add a function `ensureGoogleDriveInitialized`, and invoke it at the beginning of all exported methods. This func should check if `drive` is defined, and throws human friendly error if not.

File: lib/googleDrive/index.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
+import google from 'googleapis';
+import multiparty from 'multiparty';
+import googleAuth from './googleAuth';
+import {
+  FieldValidationError,
+} from '../tools';
+
+let drive;
+
+export function setup(googleApi = {
```
Comment (jifeon) [inline lib/googleDrive/index.js]: not sure defaults are needed here, what the purpose? Also pls add jsdoc, it should contain a note about the function must be invoked before others etc

File: lib/tools.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+/**
```
Comment (jifeon) [inline lib/tools.js]: pls rename files to `errors.js`, it was `tools.js` too long time ))

**Review by jifeon — COMMENTED** [review]
Btw, do not see your method to upload a file

**Review by jifeon — APPROVED** [review]
Looks great, comments are not critical

**Comment by jifeon** [comment]
Also it would be nice if you update README.md It's public repo btw

**Review by jifeon — APPROVED** [review]

File: lib/googleDrive/index.js (hunk 3 above)
Comment (VlVl) [reply lib/googleDrive/index.js]: the purpose - do not forget what fields are needed) ofcause the place of defaults is in jsdoc

**Review by frenzzy — APPROVED** [review]

File: lib/googleDrive/index.js (hunk 2 above)
Comment (VlVl) [inline lib/googleDrive/index.js]: Then it may be easier to call 'setup' instead of 'ensureGoogleDriveInitialized'? )

**Review by AGresvig — APPROVED** [review]
Nice work Vladimir!

**Pull request merged**